explorer/websockethandlers.go

```diff
@@ -166,6 +214,11 @@ func (exp *explorerUI) RootWebsocket(w http.ResponseWriter, r *http.Request) {
 						Extra: exp.ExtraInfo,
 					})
 					exp.NewBlockDataMtx.Unlock()
+
+					tpEvent.RLock()
+					tpEvent.Active = true
```

Comment:
Would need a read-write lock (`Lock` instead of `RLock`)

Reply:
If we use `Lock` shall we be able to activate the tpEvent?

Reply:
The mutex doesn't actually prevent us or any other goroutine from reading or writing to something.  It's only there to *allow* you to set when the data is being read or written.  In this case you want to write, so you use Lock (a R/W lock).  Where you only want to read, your can use `RLock`, the idea being you can have any number of concurrent readers, but only one writer.

I suggest to go through a tutorial on mutexs, such as https://gobyexample.com/mutexes.

But I can look again at this PR and we can resume this concurrency discussion if you think the queries are in a good enough shape.

Reply:
I think they are in good shape. Actually I've just seen that `Lock` makes sense